x/programs/examples/counter/src/lib.rs

```diff
@@ -30,3 +30,31 @@ pub fn inc(context: &mut Context, to: Address, amount: Count) -> bool {
 
     true
 }
+
+#[cfg(test)]
+#[cfg(not(feature = "bindings"))]
```

Comment:
remove before merging!
set this here because `cargo test --all` compiles this using the bindings feature flag which screws up the public functions

Reply:
I think you have to leave this, don't you?

Reply:
i was hoping I don't because then these tests don't run during CI

Reply:
Discussed offline, can fix later (what I suggested wouldn't work)